tools/c7n_gcp/c7n_gcp/resources/compute.py

```diff
@@ -144,6 +144,51 @@ def process_resource(self, client, resource):
             self.invoke_api(client, op_name, params)
 
 
+@Instance.action_registry.register('create-machine-image')
+class CreateMachineImage(MethodAction):
+    """
+    `Creates <https://cloud.google.com/compute/docs/reference/rest/beta/machineImages/insert>`_
+     Machine Image from instance.
+
+    The `name_format` specifies name of image in python `format string <https://pyformat.info/>`
+
+    Inside format string there are defined variables:
+      - `now`: current time
+      - `instance`: whole instance resource
+
+    Default name format is `{instance[name]}`
+
+    :Example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: gcp-create-machine-image
+            resource: gcp.instance
+            filters:
+              - type: value
+                key: name
+                value: instance-create-to-make-image
+            actions:
+              - type: create-machine-image
+                name_format: "{instance[name]:.50}-{now:%Y-%m-%d}"
+
+    """
+    schema = type_schema('create-machine-image', name_format={'type': 'string'})
```

Comment:
one thing that would be good to add is a label to the machine image, user supplied, but also at least one that's framework applied, the later to enable garbage collection on these images.

Reply:
I would not distinct between custodian created from normal created ones..
Our plan is to threat them as snapshots (delete after some time from creation)

Reply:
i've seen this request come in commonly for aws when using custodian for backups to distinguish the automated from any manual snapshots as an additional consideration beyond just age.

Reply:
Machine images dont have own labels (they have only instance labels, but it will restore them and also its not that straightforward to add there something)
But because users can change name of the image itself they can always add there `prefix`/`suffix` to mark thats custodian created

Reply:
ic, sounds good